android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -295,14 +295,13 @@ protected void onResume() {
       switch (data.getScheme().toLowerCase()) {
         // Android DownloadManager
         case "content":
-          checkStoragePermission(data);
-          break;
         // Chrome downloads and Filebrowsers
         case "file":
           checkStoragePermission(data);
           break;
         case "http" :
         case "https" :
+        case "keyman" :
```

Comment:
I'd prefer to refactor this so 'keyman' protocol is handled separately to 'http' and 'https'. Then have shared utility code to DRY it out if that makes sense. Otherwise the logic is a little too interleaved. (Also, we may have other uses for 'keyman:' in the future so this makes it easier if we keep it separate to begin with.